tsdb/tsm1/engine_delete_measurement.go

```diff
@@ -143,6 +147,44 @@ func (e *Engine) DeletePrefix(prefix []byte, min, max int64) error {
 		// TODO(jeff): it's also important that all of the deletes happen atomically with
 		// the deletes of the data in the tsm files.
 
+		// In this case the entire measurement (bucket) can be removed from the index.
+		if min == math.MinInt64 && max == math.MaxInt64 {
+			// Build up a set of series IDs that we need to remove from the series file.
+			set := tsdb.NewSeriesIDSet()
+			itr, err := e.index.MeasurementSeriesIDIterator(name)
+			if err != nil {
+				return err
+			}
+
+			var elem tsdb.SeriesIDElem
+			for elem, err = itr.Next(); err != nil; elem, err = itr.Next() {
+				if elem.SeriesID.IsZero() {
+					break
+				}
+
+				set.AddNoLock(elem.SeriesID)
+			}
```

Comment:
missing an error check unless Close is guaranteed to return the same one?

Reply:
Good eye!